draft-ietf-quic-recovery.md

```diff
@@ -1115,13 +1119,19 @@ acked_packet from sent_packets.
 ~~~
    InRecovery(packet_number):
      return packet_number <= end_of_recovery
+   
+   IsAppLimited():
+     bytes_in_flight >= congestion_window - kMaximumDatagramSize
```

Comment:
Isn't this backward?  Bytes in flight can't be greater than the congestion window, but if the bytes in flight are within one datagram of the window, you're clearly not limited.

Reply:
I agree this is backward. The `>=` should just be `<` or `<=`. But as far as tracking `IsAppLimited()` or whatever you want to call it, I would just suggest a modification to the code:

Instead of using `bytes_in_flight` here and the helper function `IsAppLimited`, introduce a new variable `window_full` that is only set in `OnPacketSent`. At the end of each call to that function, if `bytes_in_flight` is close enough to `congestion_window` set it to true, otherwise false. Then below, you use that variable to decide if the cwnd should be increased..

Reply:
That seems better, yes.  When you say "close enough", I assume that you mean "so close that you couldn't send anything meaningful", something like headerlen+sizeof(cid)+16+sizeof(STREAM frame overhead), so maybe sizeof(cid)+32 ?

Reply:
I was specifically not defining "close enough". I am fine with a single datagram size or something smaller.